components/brave_wallet/browser/blockchain_list_parser.cc

```diff
@@ -688,4 +688,59 @@ absl::optional<DappListMap> ParseDappLists(const std::string& json) {
   return dapp_lists;
 }
 
+absl::optional<CoingeckoIdsMap> ParseCoingeckoIdsMap(const std::string& json) {
+  // {
+  //   "0x1": {
+  //     "0xb9ef770b6a5e12e45983c5d80545258aa38f3b78": "0chain",
+  //     "0xe41d2489571d322189246dafa5ebde1f4699f498": "0x",
+  //     "0x5a3e6a77ba2f983ec0d371ea3b475f8bc0811ad5":
+  //     "0x0-ai-ai-smart-contract",
+  //     "0xfcdb9e987f9159dab2f507007d5e3d10c510aa70":
+  //     "0x1-tools-ai-multi-tool",
+  //     "0x37268c4f56ebb13dfae9c16d57d17579312d0ee1":
+  //     "0xauto-io-contract-auto-deployer"
+  //   }
+  // }
+
+  absl::optional<base::Value> records_v =
+      base::JSONReader::Read(json, base::JSON_PARSE_CHROMIUM_EXTENSIONS |
+                                       base::JSONParserOptions::JSON_PARSE_RFC);
+
+  if (!records_v || !records_v->is_dict()) {
+    VLOG(1) << "Invalid response, could not parse JSON, JSON is: " << json;
+    return absl::nullopt;
+  }
+
+  const base::Value::Dict* chain_ids = records_v->GetIfDict();
+  if (!chain_ids) {
+    return absl::nullopt;
+  }
+
+  CoingeckoIdsMap coingecko_ids_map;
+  for (const auto chain_id_record : *chain_ids) {
+    const auto& chain_id = chain_id_record.first;
+
+    if (!coingecko_ids_map.contains(chain_id)) {
+      coingecko_ids_map[chain_id] = {};
+    }
+
+    const auto* contract_addresses = chain_id_record.second.GetIfDict();
+    if (!contract_addresses) {
+      return absl::nullopt;
+    }
+
+    for (const auto contract_address_record : *contract_addresses) {
+      const auto& contract_address = contract_address_record.first;
+      const auto* coingecko_id = contract_address_record.second.GetIfString();
+      if (!coingecko_id) {
+        return absl::nullopt;
+      }
+
+      coingecko_ids_map[chain_id][contract_address] = *coingecko_id;
```

Comment:
These insertions are roughly N^2 complexity (this is how flat_map works). 
I'd suggest to use std::map for making initial container and then convert it to flat_map
https://source.chromium.org/chromium/chromium/src/+/main:base/containers/README.md;l=32

Reply:
https://github.com/brave/brave-core/pull/19489/commits/0456fc4a6ddca7a0be77d8aee8d30b6a0508658d